src/omi/dialects/oep/parser.py

```diff
@@ -10,37 +13,124 @@
+        and oemetadata specification.
 
         Parameters
         ----------
-        inp: str
-            The JSON string to be checked.
-
+        jsn
+          The dictionary to validate
+        schema
+          The jsonschema used for validation. 
+          Default is a list of all available verion specific schemas from oemetadata.
         Returns
         -------
-        bool
-            True if valid JSON, False otherwise.
+          Nothing
         """
+        self.one_schema_was_valid = False
+        # if all schemas are checked, continue until valid schema found
+        # or use latest schema that faild
+        # or detect schema version based on oemetadata metaMetadata[0].metadataVersion 
+        
+
+        #############################
+        # enable this function handle either a single schema passed as
+        # parameter or use the default (list of all available schemas).
+        # Incase one wants to test multiple metadata with various versions.
+        #############################
+        try:
+            isinstance(schemas, list)
+        except TypeError as e:
+            logging.info("Schemas must be provides as list: [schmea1, schema2, ...]")
+            raise e("parameter 'schemas' must be of type list[dict].")
+        
+        report = []
+        for schema in schemas:
+            # if not self.one_schema_was_valid:
+            # TODO: How to get the correct Validator based on the schema - dont like to use specific class
+            v = Draft7Validator(schema=schema, format_checker=FormatChecker()) 
+            v.check_schema
+        
+            for error in sorted(v.iter_errors(instance=jsn), key=str):
+                # https://python-jsonschema.readthedocs.io/en/stable/errors/#handling-validation-errors
+                error_dict = {
+                    "oemetadata schema version": schema.get("$id"),
+                    # "json path": error.json_path,
+                    "instance path": [i for i in error.absolute_path],
+                    "value that raised the error": error.instance,
+                    "error message": error.message,
+                    "schema_path": [i for i in error.schema_path],
+                }
+                report.append(error_dict)
+            # reset error collection if one schema was valid
+            # if sorted(v.iter_errors(instance=jsn), key=str) is None:
+            #     self.one_schema_was_valid = True
+            #     print(f"Valid with schema{schema.get('$id')}")
+            #     continue
+
+        create_report_json(report)
+       
+
+    def is_valid(self, inp: str or dict):
+
+        if isinstance(inp, str):
+            try:
+                jsn = json.loads(inp, encode="utf-8")
+            except ValueError:
+                return False
+        else:
+            jsn = inp
 
         try:
-            json.loads(inp)
-        except ValueError:
+            self.validate(jsn)
```

Comment:
this will not produce the expected behavior, validate should return if there was an error. Now it creates a report without raising the validation Error